### PR TITLE
Feature/#95 deepclone

### PR DESCRIPTION
**Pull request by 3timeslazy**

### CODE REVIEW

File: pkg/runtime/core/cloneable.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	ArrayType:  true,
+	ObjectType: true,
+}
+
+type Cloneable interface {
+	Value
+	Clone() Cloneable
+}
+
+func IsCloneable(value Value) bool {
```
Comment (ziflex) [inline pkg/runtime/core/cloneable.go]: Could you move this helper function to ``values`` package? It's very implementation sensitive code.

File: pkg/runtime/values/object.go [hunk 2]
```diff
@@ -202,3 +202,20 @@ func (t *Object) Remove(key String) {
 func (t *Object) SetIn(path []core.Value, value core.Value) error {
 	return SetIn(t, path, value)
 }
+
+func (t *Object) Clone() core.Cloneable {
+	cloned := NewObject()
+
+	var value core.Value
+	var keyString String
+	for _, key := range t.Keys() {
```
Comment (ziflex) [inline pkg/runtime/values/object.go]: Let's use ``t.value`` instead. Now, we get an extra iteration over inner map.

**Comment by 3timeslazy** [comment]
@ziflex done :)

File: pkg/runtime/core/cloneable.go (hunk 1 above)
Comment (3timeslazy) [reply pkg/runtime/core/cloneable.go]: What about make```IsCloneable(value values.Value) values.Boolean```?

Comment (ziflex) [reply pkg/runtime/core/cloneable.go]: @3timeslazy I do not have a strong opinion on this since it's for internal use only. 
It's up to you :)

**Pull request merged**